samples/cellular/lwm2m_carrier/src/main.c

```diff
@@ -23,14 +25,16 @@ static void lte_event_handler(const struct lte_lc_evt *const evt)
 static void on_modem_lib_init(int ret, void *ctx)
 {
 	int err;
-	static bool m_first_init = true;
 
 	ARG_UNUSED(ctx);
 
 	if (ret != 0) {
+		LOG_ERR("Modem library did not initialize: %d", ret);
 		return;
 	}
 
+	m_first_init = true;
```

Comment:
This will set the variable to true every time, it should only be set once when it is initialized.
```suggestion
```